rust/agents/relayer/src/settings/whitelist.rs

```diff
@@ -0,0 +1,209 @@
+use std::fmt;
+use std::fmt::Formatter;
+use std::marker::PhantomData;
+use std::num::ParseIntError;
+
+use ethers::types::H256;
+use serde::de::{Error, SeqAccess, Visitor};
+use serde::{Deserialize, Deserializer};
+
+use abacus_core::AbacusMessage;
+
+/// Whitelist defining which messages should be relayed. If no wishlist is provided ALL
+/// messages will be relayed.
+///
+/// Valid options for each of the tuple elements are
+/// - wildcard "*"
+/// - single value in decimal or hex (must start with `0x`) format
+/// - list of values in decimal or hex format
+///
+/// 4-tuple in the form `(sourceDomain, sourceAddress, destinationDomain, destinationAddress)`.
+#[derive(Debug, Deserialize, Default, Clone)]
+#[serde(transparent)]
+pub struct Whitelist(Option<Vec<WhitelistElement>>);
+
+#[derive(Debug, Clone)]
+enum Filter<T> {
+    Wildcard,
+    Enumerated(Vec<T>),
+}
+
+impl<T: PartialEq> Filter<T> {
+    fn matches(&self, v: &T) -> bool {
+        match self {
+            Filter::Wildcard => true,
+            Filter::Enumerated(list) => list.iter().any(|i| i == v),
+        }
+    }
+}
+
+#[derive(Deserialize)]
+#[serde(untagged)]
+enum StrOrInt<'a> {
+    Str(&'a str),
+    Int(u32),
```

Comment:
Happy either way but wondering if Uint is more accurate

Reply:
It's private, maybe, but, I don't really care that much since it is used only in the one file. Also 3-letters makes it symmetrical 😁